c/authService.c

```diff
@@ -173,16 +188,208 @@ static int serveAuthCheck(HttpService *service, HttpResponse *res) {
+    }
+    if ((strcmp(pluginID, "ORG.ZOWE.CONFIGJS") == 0) && (strcmp(serviceName, "DATA") == 0))
+    {
+      strcpy(type, "config");
+      strcpy(pluginID, subUrl[0]);
+      strcpy(scope, subUrl[1]);
+      
+    } else {
+      strcpy(type, "service");
+    }
+    char* ch; 
+    char* chReplace;
+    ch = ".";
+    chReplace = "_";
+    for (index = 0; index <= strlen(pluginID); index++)
+  	{
+  		if (pluginID[index] == *ch)  
+      {
+        pluginID[index] = *chReplace;
+      }
+    }
+    zowelog(NULL, LOG_COMP_ID_SECURITY, ZOWE_LOG_DEBUG2,
+           "RegEx match OK.");
+  }
+  else {
+    zowelog(NULL, LOG_COMP_ID_SECURITY, ZOWE_LOG_WARNING,
+           "RegEx match failed.");
+  }
+  strcpy(profileName, makeProfileName(type, 
+    productCode, 
+    instanceID, 
+    pluginID, 
+    rootServiceName,
+    serviceName,
+    method,
+    scope,
+    subUrl));
+  // printf("\n\nFinal query profileName & URL %s - %s\n\n", profileName, url);
+  
+  /* Free memory allocated to the pattern buffer by regcomp() */
+  regfree(&regex);
+  return profileName;
+}
+
+const char* makeProfileName(
+  char *type,
+  char *productCode, 
+  int instanceID, 
+  char *pluginID, 
+  char *rootServiceName, 
+  char *serviceName,
+  char *method,
+  char *scope,
+  char subUrl[250][50]) {
+  char profileName[250] = "";
+  if (strcmp(productCode, "NULL") == 0) {
+    zowelog(NULL, LOG_COMP_ID_SECURITY, ZOWE_LOG_WARNING,
+           "Broken SAF query. Missing product code.");
+    return "NULL";
+  }
+  if (instanceID == -1) {
+    zowelog(NULL, LOG_COMP_ID_SECURITY, ZOWE_LOG_WARNING,
+           "Broken SAF query. Missing instance ID.");
+    return "NULL";
+  }
+  if (strcmp(method, "NULL") == 0) {
+    zowelog(NULL, LOG_COMP_ID_SECURITY, ZOWE_LOG_WARNING,
+           "Broken SAF query. Missing method.");
+    return "NULL";
+  }
+  // char someString[50] = { strcpy(*someString, type) };
+  if (strcmp(type, "service") == 0) {
+    if (strcmp(pluginID, "NULL") == 0) {
+      zowelog(NULL, LOG_COMP_ID_SECURITY, ZOWE_LOG_WARNING,
+       "Broken SAF query. Missing plugin ID.");
+      return "NULL";
+    }
+    if (strcmp(serviceName, "NULL") == 0) {
+      zowelog(NULL, LOG_COMP_ID_SECURITY, ZOWE_LOG_WARNING,
+       "Broken SAF query. Missing service name.");
+      return "NULL";
+    }
+    snprintf(profileName, 1024, "%s.%d.SVC.%s.%s.%s", productCode, instanceID, pluginID, serviceName, method);
+    return profileName;
+    
+  } else if (strcmp(type, "config") == 0) {
+    if (strcmp(pluginID, "NULL") == 0) {
+      zowelog(NULL, LOG_COMP_ID_SECURITY, ZOWE_LOG_WARNING,
+       "Broken SAF query. Missing plugin ID.");
+      return "NULL";
+    }
+    if (strcmp(scope, "NULL") == 0) {
+      zowelog(NULL, LOG_COMP_ID_SECURITY, ZOWE_LOG_WARNING,
+       "Broken SAF query. Missing scope.");
+      return "NULL";
+    }
+    snprintf(profileName, 1024, "%s.%d.CFG.%s.%s.%s", productCode, instanceID, pluginID, method, scope); 
```

Comment:
`profileName` can hold 250 chars at max, why does 1024 go here? It's a possible buffer overflow.